src/service/GPUEnergyService6DModes.tpp

```diff
@@ -227,20 +229,33 @@ public:
 
 	void configureDevice() {
 
-		if (blockSizeReduce == 0) {
+		if (blockSizeReduceRec == 0) {
 			int id;
 			cudaVerify(cudaGetDevice(&id));
 			cudaDeviceProp deviceProp;
 			cudaVerify(cudaGetDeviceProperties(&deviceProp, id));
 			size_t sharedMem = deviceProp.sharedMemPerBlock;
 			size_t pow2 = 16;
-			while (pow2* (Common_Modes::numModesLig + Common_Modes::numModesRec + 13)*sizeof(REAL) < sharedMem) {
+			while (pow2* (Common_Modes::numModesRec + 13)*sizeof(REAL) < sharedMem) {
 				pow2 *= 2;
 			}
 
-			blockSizeReduce = pow2 / 2;
+			blockSizeReduceRec = pow2 / 2;
 		}
 
+		if (blockSizeReduceLig == 0) {
```

Comment:
please outsource to another function. you are repeating yourself. What is this plenty of code doing? Determining the allowed blocksize? (could be a good name for that function)

Reply:
done